src/main/java/com/appland/appmap/process/hooks/HttpClientRequest.java

```diff
@@ -0,0 +1,31 @@
+package com.appland.appmap.process.hooks;
+
+import com.appland.appmap.output.v1.Event;
+import com.appland.appmap.record.Recorder;
+import com.appland.appmap.transform.annotations.*;
+
+import java.io.IOException;
+import java.net.HttpURLConnection;
+import java.util.List;
+import java.util.Map;
+
+@Unique("http_client_request")
+public class HttpClientRequest {
+
+    private static final Recorder recorder = Recorder.getInstance();
+
+    @HookClass(value = "java.net.HttpURLConnection")
+    public static void connect(Event event, HttpURLConnection httpURLConnection) {
+        //TODO: ReflectiveType can be used with HttpURLConnection
+        event.setHttpClientRequest(httpURLConnection.getRequestMethod(), httpURLConnection.getURL().getHost(), httpURLConnection.getURL().getProtocol());
```

Comment:
This would be easier to read if the call to `httpURLConnection.getURL()` got moved out of this statement and assigned to a local variable.

Reply:
improved code readability by moving out URL to a local variable